pkg/xray/resource_xray_repository_config.go

```diff
@@ -14,8 +14,21 @@ import (
 
 type RepoConfiguration struct {
 	// Omitempty is used because 'vuln_contextual_analysis' is not supported by self-hosted Xray installation.
-	VulnContextualAnalysis bool `json:"vuln_contextual_analysis,omitempty"`
-	RetentionInDays        int  `json:"retention_in_days,omitempty"`
+	VulnContextualAnalysis bool      `json:"vuln_contextual_analysis,omitempty"`
+	RetentionInDays        int       `json:"retention_in_days,omitempty"`
+	Exposures              Exposures `json:"exposures"`
+}
+
+type Exposures struct {
+	ScannersCategory ScannersCategory `json:"scanners_category"`
+}
+
+type ScannersCategory struct {
+	MaliciousCode bool `json:"malicious_code_scan,omitempty"`
```

Comment:
It's better not to use `omitempty` with bools.

Reply:
Can't as the API errors out if the not-applicable fields are present. The API will only accept the valid fields for the repo type. The 'better' way will be to use `map[string]bool` struct and dynamically add only the valid fields (vs using `omitempty`). This will mean we need to fetch the repo details from RT first to determine its type, then set the fields according to that.